cli/src/action/api/rbac/assignments.rs

```diff
@@ -204,6 +245,53 @@ pub fn get_assignment(
         .map(|wrapper| wrapper.assignment)
 }
 
+pub fn update_assignment(
+    base_url: &str,
+    auth: &str,
+    assignment_update: AssignmentUpdate,
+) -> Result<(), CliError> {
+    let (id_value, id_type) = assignment_update.identity.parts();
+
+    Client::new()
+        .patch(&format!("{}/authorization/assignments/{}/{}", base_url, id_type, id_value))
+        .header("SplinterProtocolVersion", RBAC_PROTOCOL_VERSION)
+        .header("Authorization", auth)
+        .json(&assignment_update)
+        .send()
+        .map_err(|err| CliError::ActionError(format!("Failed to create assignment: {}", err)))
+        .and_then(|res| {
+            let status = res.status();
+            if status.is_success() {
+                Ok(())
+            } else if status.as_u16() == 401 {
+                Err(CliError::ActionError("Not Authorized".into()))
+            } else if status.as_u16() == 404 {
+                Err(CliError::ActionError(format!(
+                    "Authorized identity {} {} does not exist",
+                    id_type, id_value,
+                )))
+            } else if status.as_u16() == 409 {
+                Err(CliError::ActionError("One or more of the roles provided does not exist".into()))
+            } else {
+                let message = res
+                    .json::<ServerError>()
+                    .map_err(|_| {
+                        CliError::ActionError(format!(
+                            "Update assignment request failed with status code '{}', but error response \
+                            was not valid",
+                            status
+                        ))
+                    })?
+                    .message;
+
+                Err(CliError::ActionError(format!(
+                    "Update to create assignment: {}",
```

Comment:
Should this be "Failed to update assignment"?

Reply:
fixed